extension_service/datastore/providers/firestore.py

```diff
@@ -0,0 +1,270 @@
+
+            if deleted >= batch_size:
+                return await delete_collection(coll_ref, batch_size)
+
+        # Check if the collections already exist; if so, delete collections
+        airports_ref = self.__client.collection("airports")
+        airports_exist = await airports_ref.limit(1).get()
+        if airports_exist:
+            await delete_collection(airports_ref)
+
+        amenities_ref = self.__client.collection("amenities")
+        amenities_exist = await amenities_ref.limit(1).get()
+        if amenities_exist:
+            await delete_collection(amenities_ref)
+
+        flights_ref = self.__client.collection("flights")
+        flights_exist = await flights_ref.limit(1).get()
+        if flights_exist:
+            await delete_collection(flights_ref)
+
+        # initialize collections
+        for airport in airports:
+            await self.__client.collection("airports").document(str(airport.id)).set(
+                {
+                    "iata": airport.iata,
+                    "name": airport.name,
+                    "city": airport.city,
+                    "country": airport.country,
+                }
+            )
+        for amenity in amenities:
+            await self.__client.collection("amenities").document(str(amenity.id)).set(
+                {
+                    "name": amenity.name,
+                    "description": amenity.description,
+                    "location": amenity.location,
+                    "terminal": amenity.terminal,
+                    "category": amenity.category,
+                    "hour": amenity.hour,
+                    "content": amenity.content,
+                    "embedding": amenity.embedding,
+                }
+            )
+        for flight in flights:
+            await self.__client.collection("flights").document(str(flight.id)).set(
+                {
+                    "airline": flight.airline,
+                    "flight_number": flight.flight_number,
+                    "departure_airport": flight.departure_airport,
+                    "arrival_airport": flight.arrival_airport,
+                    "departure_time": flight.departure_time,
+                    "arrival_time": flight.arrival_time,
+                    "departure_gate": flight.departure_gate,
+                    "arrival_gate": flight.arrival_gate,
+                }
+            )
+
+    async def export_data(
+        self,
+    ) -> tuple[list[models.Airport], list[models.Amenity], list[models.Flight]]:
+        airport_docs = self.__client.collection("airports").stream()
+        amenities_docs = self.__client.collection("amenities").stream()
+        flights_docs = self.__client.collection("flights").stream()
+
+        airports, amenities, flights = [], [], []
+
+        for doc in airport_docs:
+            airport_dict = doc.to_dict()
+            airport_dict["id"] = doc.id
+            airports.append(models.Airport.model_validate(airport_dict))
```

Comment:
nit: it's best practice to limit the scope of the variables to when you need them:

```suggestion
        airports = []
        for doc in airport_docs:
            airport_dict = doc.to_dict()
            airport_dict["id"] = doc.id
            airports.append(models.Airport.model_validate(airport_dict))
            
        amenities = []
        # ...
```